regolith/helpers/u_milestonehelper.py

```diff
@@ -0,0 +1,148 @@
+"""Helper for updating milestones to the projecta collection.
+    It can update the status and due date of a projectum.
+    It can add a new milestone to the projecta collection.
+"""
+import datetime as dt
+import sys
+from dateutil import parser
+from regolith.helpers.basehelper import DbHelperBase
+from regolith.fsclient import _id_key
+from regolith.tools import all_docs_from_collection
+
+
+TARGET_COLL = "projecta"
+
+def subparser(subpi):
+    subpi.add_argument("projectum_id", help="the id of the projectum")
+    subpi.add_argument("--number",
+                        help="number of the milestone to update from numbered list")
+    subpi.add_argument("--due_date",
+                       help="new due date of the milestone in ISO format (YYYY-MM-DD) "
+                            "required to add a new milestone")
+    subpi.add_argument("-s", "--status",
+                       help="initial of the status of the milestone/deliverable: "
+                            "p = proposed, s = started, f = finished, c = cancelled")
+    subpi.add_argument("--name",
+                       help="name of the new milestone. "
+                            "required to add a new milestone")
+    subpi.add_argument("-o", "--objective",
+                       help="objective of the new milestone. "
+                            "required to add a new milestone")
+    subpi.add_argument("-t", "--type",
+                       help="initial of type of the new milestone "
+                            "m = meeting, r = release, p = pull request, o = other")
+    # Do not delete --database arg
+    subpi.add_argument("--database",
+                       help="The database that will be updated.  Defaults to "
+                            "first database in the regolithrc.json file.")
+    return subpi
+
+class MilestoneUpdaterHelper(DbHelperBase):
+    """Helper for updating milestones to the projecta collection.
+    """
+    # btype must be the same as helper target in helper.py
+    btype = "u_milestone"
+    needed_dbs = [f'{TARGET_COLL}']
+
+    def construct_global_ctx(self):
+        """Constructs the global context"""
+        super().construct_global_ctx()
+        gtx = self.gtx
+        rc = self.rc
+        rc.coll = f"{TARGET_COLL}"
+        if not rc.database:
+            rc.database = rc.databases[0]["name"]
+        gtx[rc.coll] = sorted(
+            all_docs_from_collection(rc.client, rc.coll), key=_id_key
+        )
+        gtx["all_docs_from_collection"] = all_docs_from_collection
+        gtx["float"] = float
+        gtx["str"] = str
+        gtx["zip"] = zip
+
+    def db_updater(self):
+        rc = self.rc
+        key = rc.projectum_id
+        allowed_types = {'m': 'meeting', 'r':'release', 'p':'pr', 'o':'other'}
```

Comment:
make this a static global variable, so put it at the top of the file capitalized ``ALLOWED_TYPES``

Reply:
done!